server/rpc/yorkie_server.go

```diff
@@ -392,13 +392,27 @@ func (s *yorkieServer) WatchDocuments(
 		return err
 	}
 
+	locker, err := s.backend.Coordinator.NewLocker(
+		stream.Context(),
+		sync.NewKey(cli.ID.String()),
+	)
+	if err != nil {
+		return err
+	}
+	if err := locker.Lock(stream.Context()); err != nil {
+		return err
+	}
+
 	subscription, peersMap, err := s.watchDocs(stream.Context(), *cli, docKeys)
 	if err != nil {
 		logging.From(stream.Context()).Error(err)
 		return err
 	}
 	defer func() {
 		s.unwatchDocs(docKeys, subscription)
+		if err := locker.Unlock(stream.Context()); err != nil {
+			logging.DefaultLogger().Error(err)
+		}
```

Comment:
1. If an error is returned by `s.watchDocs`, releasing the lock is not executed. So, How about moving `locker.Unlock` to just below `locker.lock` with defer?

2. `locker.Unlock` may not be executed when the stream is terminated and the context is in the cancel state. How about using `context.Background` instead of `stream.Context`?

3. We need to separate the key of the lock with `snapshot-*` or `pushpull-*`. How about creating a key with the prefix `watchdocs-*`?